RecoPixelVertexing/PixelTriplets/interface/HitTripletGeneratorFromPairAndLayers.h

```diff
@@ -33,6 +33,17 @@ class HitTripletGeneratorFromPairAndLayers {
                             const edm::Event & ev, const edm::EventSetup& es,
                             const SeedingLayerSetsHits::SeedingLayerSet& pairLayers,
                             const std::vector<SeedingLayerSetsHits::SeedingLayer>& thirdLayers) = 0;
+
+    virtual void hitTriplets(
+	const TrackingRegion& region, 
+	OrderedHitTriplets & result,
+	const edm::EventSetup & es,
+	const HitDoublets & doublets,
+	const RecHitsSortedInPhi ** thirdHitMap,
+	const std::vector<const DetLayer *> & thirdLayerDetLayer,
+	const int nThirdLayers)
+	{};
```

Comment:
Are you sure you don't want this to be pure virtual (`=0`)? OK, it would mean that you would have to implement this in a couple of other triplet generators too, but then we would notice already at compile-time if some triplet generator would not implement it (as opposed to wondering unexpected results).